### PR TITLE
Initial development

### PR DESCRIPTION
**Pull request by gerryster**
This adds our first [CircleCI Orb](https://circleci.com/orbs/) for posting job status to an MS Teams channel. Note that I have only published it to the Orbs development channel. I would like to run it in the wild for a few days before publishing a version 1.0 to the production Orbs channel. I plan on doing that before this is merged to master.

### CODE REVIEW

File: src/orbs/status_to_ms_teams_pure_bash.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                }
+              ]
+            }
+            END_HEREDOC
+            )
+            echo "$MS_TEAMS_MSG_TEMPLATE" > /tmp/ms_teams_message
+
+            echo begin teams message
+            cat /tmp/ms_teams_message
+            echo end teams message
```
Comment (roeyby) [inline src/orbs/status_to_ms_teams_pure_bash.yml]: As I understand it, on_fail triggers if a previous step returned a non-zero value. But here the last command is an echo which resets the return value to zero. Will any error in the run terminate it prematurely and return a non-zero value? I'm not sure how this works so I'm raising it as a possible issue.

File: test/run [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o xtrace
+
+# Validate all orbs:
+# The path modification is needed for CI as a newer version of circleci is
```
[the remark this answers is not in the document]
Comment (gerryster) [reply test/run]: Good call. I'll do that.

File: src/orbs/status_to_ms_teams_pure_bash.yml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+      - run:
+          name: Report Build Success
+          command: |
+            MSG_PATH=/tmp/ms_teams_success_message
```
[the remark this answers is not in the document]
Comment (gerryster) [reply src/orbs/status_to_ms_teams_pure_bash.yml]: Yea, I thought about that too as it is more secure and preferable to create unique temp files. There is apparently a standard Unix command, mktemp, which does exactly that. I did however run up against a limitation. I am not unsure of how to save state (other than on the file system or some external database) across multiple run steps. I really wanted to find a way to persist a variable across multiple steps. One way around this is to pass in parameters or environment variables to the entire orb. However, I don't want to have to burden the consumer of the orb with that implementation detail. Let me know if you have other ideas on how to accomplish this. I'd also be happy to go into more detail on this in call.

File: src/orbs/status_to_ms_teams_pure_bash.yml (hunk 1 above)
Comment (gerryster) [reply src/orbs/status_to_ms_teams_pure_bash.yml]: Good call. Yea, this was really debugging code which should be removed to avoid any heisenbugs.

**Comment by gerryster** [comment]
Back to you @roeyby. Let me know if you want to hop on a call to discuss the temp file situation.

**Comment by roeyby** [comment]
Approved

**Pull request merged**